### ISSUE TITLE
Named export renaming uses reserved keywords

### ISSUE THREAD

**Issue by FruitieX** [rollup/rollup#2761]
- Rollup Version: v1.7.0
- Operating System (or Browser): Linux / Firefox v64
- Node Version: v10.12.0

Rollup seems to rename named exports in some cases when using code splitting (`export { var1 as a, var2 as b, ... }`, see first Rollup repl link for minimal repro).

This is all fine, but issues arise if one of the chunks contains a huge amount of named exports (in my case, a UI library with icons). After we run out of single letter tokens, Rollup will start using, `aa, ab ...` and eventually `da, db ... do`. It seems there's no checks in place that avoid using reserved keywords here such as `do`, which now occurs in my project. My bundle ends up containing:

```
import { a as firstExport, b as secondExport, ... dn as IconFavorite, do as IconFavoriteFilled, dp as IconFilledSquare, ... }
                                                                      ^^
```

which create-react-app chokes on when importing the resulting bundle (`Unexpected keyword 'do'`).

#### Minimal reproduction of the renaming behavior
[repl link](https://rollupjs.org/repl?version=1.7.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBmcm9tJTIwJy4lMkZjb25zdGFudHMnJTNCJTVDbmV4cG9ydCUyMGNvbnN0JTIwbGF6eSUyMCUzRCUyMGltcG9ydCgnLiUyRmxhenknKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmNvbnN0YW50cy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBjb25zdCUyMGJsYWNrJTIwJTNEJTIwJyUyMzAwMDAwMCclM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybGF6eS5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAlN0IlMjBibGFjayUyMCU3RCUyMGZyb20lMjAnLiUyRmNvbnN0YW50cyclM0IlNUNuZXhwb3J0JTIwZGVmYXVsdCUyMCgpJTIwJTNEJTNFJTIwYmxhY2slM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXNtJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

Note how `black` is renamed to `a` in the above example.

#### Minimal reproduction of the rename `as do` problem
[repl link](https://rollupjs.org/repl?version=1.7.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBmcm9tJTIwJy4lMkZjb25zdGFudHMnJTNCJTVDbmV4cG9ydCUyMGNvbnN0JTIwbGF6eSUyMCUzRCUyMGltcG9ydCgnLiUyRmxhenknKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmNvbnN0YW50cy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjB2YXIlMjB2MSUyQ3YyJTJDdjMlMkN2NCUyQ3Y1JTJDdjYlMkN2NyUyQ3Y4JTJDdjklMkN2MTAlMkN2MTElMkN2MTIlMkN2MTMlMkN2MTQlMkN2MTUlMkN2MTYlMkN2MTclMkN2MTglMkN2MTklMkN2MjAlMkN2MjElMkN2MjIlMkN2MjMlMkN2MjQlMkN2MjUlMkN2MjYlMkN2MjclMkN2MjglMkN2MjklMkN2MzAlMkN2MzElMkN2MzIlMkN2MzMlMkN2MzQlMkN2MzUlMkN2MzYlMkN2MzclMkN2MzglMkN2MzklMkN2NDAlMkN2NDElMkN2NDIlMkN2NDMlMkN2NDQlMkN2NDUlMkN2NDYlMkN2NDclMkN2NDglMkN2NDklMkN2NTAlMkN2NTElMkN2NTIlMkN2NTMlMkN2NTQlMkN2NTUlMkN2NTYlMkN2NTclMkN2NTglMkN2NTklMkN2NjAlMkN2NjElMkN2NjIlMkN2NjMlMkN2NjQlMkN2NjUlMkN2NjYlMkN2NjclMkN2NjglMkN2NjklMkN2NzAlMkN2NzElMkN2NzIlMkN2NzMlMkN2NzQlMkN2NzUlMkN2NzYlMkN2NzclMkN2NzglMkN2NzklMkN2ODAlMkN2ODElMkN2ODIlMkN2ODMlMkN2ODQlMkN2ODUlMkN2ODYlMkN2ODclMkN2ODglMkN2ODklMkN2OTAlMkN2OTElMkN2OTIlMkN2OTMlMkN2OTQlMkN2OTUlMkN2OTYlMkN2OTclMkN2OTglMkN2OTklMkN2MTAwJTJDdjEwMSUyQ3YxMDIlMkN2MTAzJTJDdjEwNCUyQ3YxMDUlMkN2MTA2JTJDdjEwNyUyQ3YxMDglMkN2MTA5JTJDdjExMCUyQ3YxMTElMkN2MTEyJTJDdjExMyUyQ3YxMTQlMkN2MTE1JTJDdjExNiUyQ3YxMTclMkN2MTE4JTJDdjExOSUyQ3YxMjAlMkN2MTIxJTJDdjEyMiUyQ3YxMjMlMkN2MTI0JTJDdjEyNSUyQ3YxMjYlMkN2MTI3JTJDdjEyOCUyQ3YxMjklMkN2MTMwJTJDdjEzMSUyQ3YxMzIlMkN2MTMzJTJDdjEzNCUyQ3YxMzUlMkN2MTM2JTJDdjEzNyUyQ3YxMzglMkN2MTM5JTJDdjE0MCUyQ3YxNDElMkN2MTQyJTJDdjE0MyUyQ3YxNDQlMkN2MTQ1JTJDdjE0NiUyQ3YxNDclMkN2MTQ4JTJDdjE0OSUyQ3YxNTAlMkN2MTUxJTJDdjE1MiUyQ3YxNTMlMkN2MTU0JTJDdjE1NSUyQ3YxNTYlMkN2MTU3JTJDdjE1OCUyQ3YxNTklMkN2MTYwJTJDdjE2MSUyQ3YxNjIlMkN2MTYzJTJDdjE2NCUyQ3YxNjUlMkN2MTY2JTJDdjE2NyUyQ3YxNjglMkN2MTY5JTJDdjE3MCUyQ3YxNzElMkN2MTcyJTJDdjE3MyUyQ3YxNzQlMkN2MTc1JTJDdjE3NiUyQ3YxNzclMkN2MTc4JTJDdjE3OSUyQ3YxODAlMkN2MTgxJTJDdjE4MiUyQ3YxODMlMkN2MTg0JTJDdjE4NSUyQ3YxODYlMkN2MTg3JTJDdjE4OCUyQ3YxODklMkN2MTkwJTJDdjE5MSUyQ3YxOTIlMkN2MTkzJTJDdjE5NCUyQ3YxOTUlMkN2MTk2JTJDdjE5NyUyQ3YxOTglMkN2MTk5JTJDdjIwMCUyQ3YyMDElMkN2MjAyJTJDdjIwMyUyQ3YyMDQlMkN2MjA1JTJDdjIwNiUyQ3YyMDclMkN2MjA4JTJDdjIwOSUyQ3YyMTAlMkN2MjExJTJDdjIxMiUyQ3YyMTMlMkN2MjE0JTJDdjIxNSUyQ3YyMTYlMkN2MjE3JTJDdjIxOCUyQ3YyMTklMkN2MjIwJTJDdjIyMSUyQ3YyMjIlMkN2MjIzJTJDdjIyNCUyQ3YyMjUlMkN2MjI2JTJDdjIyNyUyQ3YyMjglMkN2MjI5JTJDdjIzMCUyQ3YyMzElMkN2MjMyJTJDdjIzMyUyQ3YyMzQlMkN2MjM1JTJDdjIzNiUyQ3YyMzclMkN2MjM4JTJDdjIzOSUyQ3YyNDAlMkN2MjQxJTJDdjI0MiUyQ3YyNDMlMkN2MjQ0JTJDdjI0NSUyQ3YyNDYlMkN2MjQ3JTJDdjI0OCUyQ3YyNDklMkN2MjUwJTJDdjI1MSUyQ3YyNTIlMkN2MjUzJTJDdjI1NCUyQ3YyNTUlMkN2MjU2JTJDdjI1NyUyQ3YyNTglMkN2MjU5JTJDdjI2MCUyQ3YyNjElMkN2MjYyJTJDdjI2MyUyQ3YyNjQlMkN2MjY1JTJDdjI2NiUyQ3YyNjclMkN2MjY4JTJDdjI2OSUyQ3YyNzAlMkN2MjcxJTJDdjI3MiUzQiU1Q24lMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybGF6eS5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAlN0IlMjB2MSUyMCU3RCUyMGZyb20lMjAnLiUyRmNvbnN0YW50cyclM0IlNUNuZXhwb3J0JTIwZGVmYXVsdCUyMCgpJTIwJTNEJTNFJTIwdjElM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXNtJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

With a huge number of variables (271 to be exact :smile:) we can see the problematic output being generated. ctrl+f search for `as do` or `do as` in the output and you will spot the problem, which also the repl's syntax highlighting displays as a keyword:

![image](https://user-images.githubusercontent.com/3673134/54695755-32d8e000-4b33-11e9-8b2f-1706eaf57af5.png)

### Expected Behavior

Avoid using reserved keywords when renaming named exports

### Actual Behavior

Reserved keywords such as `do` are being used for renamed named exports.

**Comment by FruitieX** [comment]
Oops looks like the second repl link is broken due to being too long. Here's a version with shorter variable names: [repl link](https://rollupjs.org/repl?version=1.7.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBmcm9tJTIwJy4lMkZjb25zdGFudHMnJTNCJTVDbmV4cG9ydCUyMGNvbnN0JTIwbGF6eSUyMCUzRCUyMGltcG9ydCgnLiUyRmxhenknKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmNvbnN0YW50cy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjB2YXIlMjB2MSUyQ3YyJTJDdjMlMkN2NCUyQ3Y1JTJDdjYlMkN2NyUyQ3Y4JTJDdjklMkN2MTAlMkN2MTElMkN2MTIlMkN2MTMlMkN2MTQlMkN2MTUlMkN2MTYlMkN2MTclMkN2MTglMkN2MTklMkN2MjAlMkN2MjElMkN2MjIlMkN2MjMlMkN2MjQlMkN2MjUlMkN2MjYlMkN2MjclMkN2MjglMkN2MjklMkN2MzAlMkN2MzElMkN2MzIlMkN2MzMlMkN2MzQlMkN2MzUlMkN2MzYlMkN2MzclMkN2MzglMkN2MzklMkN2NDAlMkN2NDElMkN2NDIlMkN2NDMlMkN2NDQlMkN2NDUlMkN2NDYlMkN2NDclMkN2NDglMkN2NDklMkN2NTAlMkN2NTElMkN2NTIlMkN2NTMlMkN2NTQlMkN2NTUlMkN2NTYlMkN2NTclMkN2NTglMkN2NTklMkN2NjAlMkN2NjElMkN2NjIlMkN2NjMlMkN2NjQlMkN2NjUlMkN2NjYlMkN2NjclMkN2NjglMkN2NjklMkN2NzAlMkN2NzElMkN2NzIlMkN2NzMlMkN2NzQlMkN2NzUlMkN2NzYlMkN2NzclMkN2NzglMkN2NzklMkN2ODAlMkN2ODElMkN2ODIlMkN2ODMlMkN2ODQlMkN2ODUlMkN2ODYlMkN2ODclMkN2ODglMkN2ODklMkN2OTAlMkN2OTElMkN2OTIlMkN2OTMlMkN2OTQlMkN2OTUlMkN2OTYlMkN2OTclMkN2OTglMkN2OTklMkN2MTAwJTJDdjEwMSUyQ3YxMDIlMkN2MTAzJTJDdjEwNCUyQ3YxMDUlMkN2MTA2JTJDdjEwNyUyQ3YxMDglMkN2MTA5JTJDdjExMCUyQ3YxMTElMkN2MTEyJTJDdjExMyUyQ3YxMTQlMkN2MTE1JTJDdjExNiUyQ3YxMTclMkN2MTE4JTJDdjExOSUyQ3YxMjAlMkN2MTIxJTJDdjEyMiUyQ3YxMjMlMkN2MTI0JTJDdjEyNSUyQ3YxMjYlMkN2MTI3JTJDdjEyOCUyQ3YxMjklMkN2MTMwJTJDdjEzMSUyQ3YxMzIlMkN2MTMzJTJDdjEzNCUyQ3YxMzUlMkN2MTM2JTJDdjEzNyUyQ3YxMzglMkN2MTM5JTJDdjE0MCUyQ3YxNDElMkN2MTQyJTJDdjE0MyUyQ3YxNDQlMkN2MTQ1JTJDdjE0NiUyQ3YxNDclMkN2MTQ4JTJDdjE0OSUyQ3YxNTAlMkN2MTUxJTJDdjE1MiUyQ3YxNTMlMkN2MTU0JTJDdjE1NSUyQ3YxNTYlMkN2MTU3JTJDdjE1OCUyQ3YxNTklMkN2MTYwJTJDdjE2MSUyQ3YxNjIlMkN2MTYzJTJDdjE2NCUyQ3YxNjUlMkN2MTY2JTJDdjE2NyUyQ3YxNjglMkN2MTY5JTJDdjE3MCUyQ3YxNzElMkN2MTcyJTJDdjE3MyUyQ3YxNzQlMkN2MTc1JTJDdjE3NiUyQ3YxNzclMkN2MTc4JTJDdjE3OSUyQ3YxODAlMkN2MTgxJTJDdjE4MiUyQ3YxODMlMkN2MTg0JTJDdjE4NSUyQ3YxODYlMkN2MTg3JTJDdjE4OCUyQ3YxODklMkN2MTkwJTJDdjE5MSUyQ3YxOTIlMkN2MTkzJTJDdjE5NCUyQ3YxOTUlMkN2MTk2JTJDdjE5NyUyQ3YxOTglMkN2MTk5JTJDdjIwMCUyQ3YyMDElMkN2MjAyJTJDdjIwMyUyQ3YyMDQlMkN2MjA1JTJDdjIwNiUyQ3YyMDclMkN2MjA4JTJDdjIwOSUyQ3YyMTAlMkN2MjExJTJDdjIxMiUyQ3YyMTMlMkN2MjE0JTJDdjIxNSUyQ3YyMTYlMkN2MjE3JTJDdjIxOCUyQ3YyMTklMkN2MjIwJTJDdjIyMSUyQ3YyMjIlMkN2MjIzJTJDdjIyNCUyQ3YyMjUlMkN2MjI2JTJDdjIyNyUyQ3YyMjglMkN2MjI5JTJDdjIzMCUyQ3YyMzElMkN2MjMyJTJDdjIzMyUyQ3YyMzQlMkN2MjM1JTJDdjIzNiUyQ3YyMzclMkN2MjM4JTJDdjIzOSUyQ3YyNDAlMkN2MjQxJTJDdjI0MiUyQ3YyNDMlMkN2MjQ0JTJDdjI0NSUyQ3YyNDYlMkN2MjQ3JTJDdjI0OCUyQ3YyNDklMkN2MjUwJTJDdjI1MSUyQ3YyNTIlMkN2MjUzJTJDdjI1NCUyQ3YyNTUlMkN2MjU2JTJDdjI1NyUyQ3YyNTglMkN2MjU5JTJDdjI2MCUyQ3YyNjElMkN2MjYyJTJDdjI2MyUyQ3YyNjQlMkN2MjY1JTJDdjI2NiUyQ3YyNjclMkN2MjY4JTJDdjI2OSUyQ3YyNzAlMkN2MjcxJTJDdjI3MiUzQiU1Q24lMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybGF6eS5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAlN0IlMjB2MSUyMCU3RCUyMGZyb20lMjAnLiUyRmNvbnN0YW50cyclM0IlNUNuZXhwb3J0JTIwZGVmYXVsdCUyMCgpJTIwJTNEJTNFJTIwdjElM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXNtJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

**Comment by lukastaegert** [comment]
Thanks for the issue. There is in fact sanitation for regular variable names but not for auto-shortened exports and imports. Should not be too difficult to fix. I am working on a few other things at the moment but it should be possible to sneak in a fix for this soon.

**Comment by lukastaegert** [comment]
As it turns out, it was really not a lot of effort as I could just re-use your nice test case as an official test and we already have a table of reserved names in the code. Fix at #2768.
